proto/minder/v1/minder.proto

```diff
@@ -3411,3 +3412,139 @@ message UpstreamEntityRef {
+        // templates that can be parameterized with variables. Parametrization
+        // is done using RFC 6570.
+        string endpoint = 1 [
+            (buf.validate.field).string = {
+                pattern: "^https?://.*$",
+                max_len: 800,
+            },
+            (google.api.field_behavior) = REQUIRED
+        ];
+
+        // method is the HTTP method to use for the request.
+        // If left unset, it will default to "GET".
+        string method = 2 [
+            (buf.validate.field).string = {
+                in: ["GET", "POST", "PUT", "PATCH", "DELETE"]
+            },
+            (buf.validate.field).ignore = IGNORE_IF_DEFAULT_VALUE
+        ];
+
+        // headers is a map of headers to send with the request.
+        map<string, string> headers = 3;
+
+        oneof body {
+            // body is the body of the request.
+            google.protobuf.Struct bodyobj = 4;
+            // bodystr is the body of the request as a string.
+            string bodystr = 5 [
+                (buf.validate.field).string = {
+                    max_len: 1000,
+                }
+            ];
+        }
+
+        // parse is the parse configuration for the response.
+        // This allows us to serialize the response into a structured format,
+        // or not.
+        // If left unset, the response will be treated as a string.
+        // If set to "json", the response will be parsed as JSON.
+        string parse = 6 [
+            (buf.validate.field).string = {
+                in: ["json"]
+            },
+            (buf.validate.field).ignore = IGNORE_IF_DEFAULT_VALUE
+        ];
+
+        message Fallback {
+            int32 http_status = 1 [
+                (buf.validate.field).int32 = {gte: 100, lte: 599},
+                (buf.validate.field).ignore = IGNORE_IF_DEFAULT_VALUE
+            ];
+
+            string body = 2 [
+                (buf.validate.field).string = {
+                    max_len: 1000,
+                },
+                (buf.validate.field).ignore = IGNORE_IF_DEFAULT_VALUE
+            ];
+        }
+
+        // fallback is the fallback configuration for the response in case
+        // of an unexpected status code.
+        repeated Fallback fallback = 7;
+
+        // expected_status is the expected status code for the response.
+        // This may be repeated to allow for multiple expected status codes.
+        // If left unset, it will default to 200.
+        repeated int32 expected_status = 8 [
+            (buf.validate.field).repeated = {
+                items: {
+                    int32: {
+                        gte: 100,
+                        lte: 599,
+                    }
+                }
+            },
+            (buf.validate.field).ignore = IGNORE_IF_DEFAULT_VALUE
+        ];
+    }
+
+    // defs is the list of definitions for the REST API.
+    map<string, Def> defs = 1;
```

Comment:
@blkt this is the nesting. It'll end up like:

```
rests:
  def:
    foo:
       <endpoint config>...
    bar:
       ...
```

Reply:
I think that's OK, it also solves the problem of where to put extra fields specific to the `rest(s)` and/or other types.